src/chunks/storage/quantized.rs

```diff
@@ -80,23 +80,23 @@ impl QuantizedArray {
             None
         };
 
-        let mut quantizers = Vec::with_capacity(quantized_len);
-        for _ in 0..quantized_len {
-            let mut subquantizer_vec =
-                vec![0f32; n_centroids * (reconstructed_len / quantized_len)];
-            read.read_f32_into::<LittleEndian>(&mut subquantizer_vec)
-                .map_err(|e| ErrorKind::io_error("Cannot read subquantizer", e))?;
-            let subquantizer = Array2::from_shape_vec(
-                (n_centroids, reconstructed_len / quantized_len),
-                subquantizer_vec,
-            )
-            .map_err(Error::Shape)?;
-            quantizers.push(subquantizer);
-        }
+        let quantizer_shape = (
+            quantized_len,
+            n_centroids,
+            reconstructed_len / quantized_len,
```

Comment:
Is there anything to verify that `reconstructed_len % quantized_len == 0`?

Reply:
`reductive` has quite some assertions. This one is not relevant to reductive outside training, since reductive just gets a tensor, that determines the quantizer parameters and the PQ constructor verifies that the projection matrix has the corresponding size.

But there could be malformed files, so we don't want to abort with a failed assertion, but instead return `Err`. I'll make an issue and do a separate PR.

Reply:
Given that's really an edge-case, but since we're returning `Result` here anyways...

Reply:
Cosmic ray that bit flips memory while reading/writing the embeddings? ;)

At any rate, I have opened #55 for this. Good catch.